src/hooks/useAdoptiumContributorsApi.tsx

```diff
@@ -117,12 +119,15 @@ async function fetchRandomContributor() {
         Math.floor(Math.random() * Math.floor(maxContributors)) + 1
       );
     }
-    const [randomPage, lastPage] = await getMaxContributors();
+    let [randomPage, lastPage] = await getMaxContributors();
 
-    let contributor = await getContributor(randomPage);
-    while (excludedContributors.includes(contributor.login)) {
+    let contributor: Contributor | null = null;
+    do {
       contributor = await getContributor(randomPage);
+      // Get a new random page incase we initially fetch a bot user
+      [randomPage, lastPage] = await getMaxContributors();
     }
+    while (excludedContributors.includes(contributor.login))
```

Comment:
Isn't this better written as:

```suggestion
    let contributor: Contributor | null = null;
    do {
      let [randomPage, lastPage] = await getMaxContributors();
      contributor = await getContributor(randomPage);
    }
    while (excludedContributors.includes(contributor.login))
```
That will avoid getting the contributor twice when successful first time.